meta-client/src/test/java/org/apache/helix/metaclient/impl/zk/TestZkMetaClient.java

```diff
@@ -334,6 +320,35 @@ public void handleDirectChildChange(String key) throws Exception {
     }
   }
 
+  @Test
+  public void testConnectStateChangeListener() throws Exception {
```

Comment:
neat : good to add few comments above this test what all test scenarios it's covering. Once it covers all scenarios it's obvious so no comment is needed but until then reader might not understand what is covered and what is left.

Reply:
Next PR will cover other stages. Won't have gap in between releases.